src/osd/PG.cc

```diff
@@ -6545,7 +6549,6 @@ boost::statechart::result
 PG::RecoveryState::RepWaitBackfillReserved::react(const RemoteReservationRejected &evt)
 {
   PG *pg = context< RecoveryMachine >().pg;
-  pg->reject_reservation();
```

Comment:
We've just switched from a single call site to 4, and a new invariant where any time you post a RemoteReservationRejected request you have to also call this function. That's not maintainable and is basically hacking the state machine to bits. These are separate states and so should be different transitions.

Reply:
Yeah, let me take another look at this.